### DISCUSSION TITLE
[Q] Corrupted zip: can't find end of central directory

### DISCUSSION THREAD

**Discussion by rrib** [exceljs/exceljs#2388] [Q&A]
Hi, 
I´m trying to read a xlxs file and we have the next error:
       **Error: Corrupted zip: can't find end of central directory**

   const workbook = new Excel.Workbook();
    await workbook.xlsx.read(stream)
    const sheet = workbook.getWorksheet('Accounts')   // the error procuces here

**Comment by intervalia** [comment]
I ran into this today as well.
I had an XLS file.
I resaved it as an XLSX file
And I was able to read that just fine.

It seems that XLS files are not compresses and XLSX files are.
It looks like the code is ALWAYS attempting to decompress the file.

Please try resaving your file as XLSX and see if that file will work. If it does then we know that we need a way to disable the decompression.

**Comment by lvelizh-dev** [comment]
@intervalia I ran into the same problem, so I changed the file extension when saving it from XLS to XLSX and now it is working.
Thank you!

**Comment by daveclay** [comment]
Are you assuming the people writing the code are the same people creating the Excel files? Can't the library detect which files are zipped or not, especially given the `xls` vs `xlsx` format definitions?

**Comment by iiian** [comment]
I am currently encountering this issue unexpectedly in production and don't know how to fix it. It started happening out of nowhere, I think it might have been caused by switching to Typescript ^5.0?
